### DISCUSSION TITLE
request Snapshot Feature

### DISCUSSION THREAD

**Discussion by fredbjer** [juicedata/juicefs#1725] [General]
What's the plan of 'snapshot' feature in the community?  Is it has a design now?

**Comment by SandyXSD** [comment]
Sorry but currently there is no plan for this feature.

**Reply by solracsf** [reply]
@SandyXSD it is mentioned on the Roadmap https://github.com/juicedata/juicefs/blob/main/README.md#roadmap

**Reply by dantefromhell** [reply]
This mysql user article (https://juicefs.com/docs/cloud/backup_mysql_in_juicefs/) mentions the `juicefs snapshot` command. The last update to the article was Dec 15, 2022.

I'm wondering if there's current work that happened on juicefs regarding the snapshot feature?

**Reply by SandyXSD** [reply]
This article is for JuiceFS cloud service edition.

**Reply by ak2k** [reply]
@SandyXSD Hi, I would absolutely love the feature. 

The design of juicefs would seem to make implementing a snapshot feature straightforward. 

To the extent that you're opposed to adding the feature to community because you intend for it to by a premium feature, perhaps it's worth updating the roadmap in the readme to clarify that this isn't for the community edition. 

If you're open to it but haven't gotten around to it, would you be open to a contribution via a pull request?
